app/controllers/shoutouts_controller.rb

```diff
@@ -32,7 +32,10 @@ def create
       end
     end
 
-    if params[:text].include?(params[:user_name])
+    recipients = params[:text]
+                    .scan(/(@(\w+|\.)+)/)
+                    .map(&:first).map { |r| r[1..-1] }
+    if (recipients.include?(params[:user_name]))
```

Comment:
Now we've extracted recipients we can use it later on on line 49 instead of doing the work twice.  I'll approve on the assumption you're gonna do that, so you can merge when you're ready.